web/react/components/create_post.jsx

```diff
@@ -112,6 +131,63 @@ module.exports = React.createClass({
     handleUserInput: function(messageText) {
         this.resizePostHolder();
         this.setState({messageText: messageText});
+
+        // look to see if the message begins with any carets to indicate that it's a reply
+        var replyMatch = messageText.match(/^\^+/g);
+        if (replyMatch) {
+            // the number of carets indicates how many message threads back we're replying to
```

Comment:
Should we keep state of how many `^` we find and check that here? That way we don't have to execute below if the number of `^` has not changed.